tests/test_run.py

```diff
@@ -1,49 +1,97 @@
 #!/usr/bin/env python3
 
 import subprocess
+import shutil
+from pathlib import Path
+
+import pytest
 
 MASTER_PASSWORD = 'test'
-HEADER = 'url,username,password\n'
-IMPORT_CREDENTIAL = 'http://www.example.com,foo,bar\n'
-EXPECTED_EXPORT_OUTPUT = f'{HEADER}http://www.stealmylogin.com,test,test\n'
-EXPECTED_IMPORT_OUTPUT = EXPECTED_EXPORT_OUTPUT + IMPORT_CREDENTIAL
+HEADER = 'url,username,password'
+IMPORT_CREDENTIAL = 'https://www.example.com,foo,bar'
+EXPECTED_EXPORT_OUTPUT = [HEADER, 'http://www.stealmylogin.com,test,test']
+EXPECTED_IMPORT_OUTPUT = EXPECTED_EXPORT_OUTPUT + [IMPORT_CREDENTIAL]
+
+
+@pytest.fixture
+def clean_profile(tmp_path):
+    """
+    Copies the requested profile to a temporary directory and returns
+    the path to the new copy.
+    """
+    def _setup(profile_name):
+        src = Path('tests') / profile_name
+        dst = tmp_path / profile_name
+        shutil.copytree(src, dst)
+        return dst
+    return _setup
 
 
-def run_ffpass(mode, path):
-    command = ["ffpass", mode, "-d", path]
+def run_ffpass_cmd(mode, path):
+    command = ["python", "./ffpass/__init__.py", mode, "--debug", "--dir", str(path)]
+
     if mode == 'import':
-        ffpass_input = HEADER + IMPORT_CREDENTIAL
+        ffpass_input = "\n".join([HEADER, IMPORT_CREDENTIAL])
     else:
         ffpass_input = None
 
     return subprocess.run(command, stdout=subprocess.PIPE, input=ffpass_input, encoding='utf-8')
 
 
-def test_legacy_firefox_export():
-    r = run_ffpass('export', 'tests/firefox-70')
+def stdout_splitter(input_text):
+    return [x for x in input_text.splitlines()]
+
+
+def test_legacy_firefox_export(clean_profile):
+    r = run_ffpass_cmd('export', clean_profile('firefox-70'))
     r.check_returncode()
-    assert r.stdout == EXPECTED_EXPORT_OUTPUT
+    actual_export_output = stdout_splitter(r.stdout)
+    assert actual_export_output == EXPECTED_EXPORT_OUTPUT
 
 
-def test_firefox_export():
-    r = run_ffpass('export', 'tests/firefox-84')
+def test_firefox_export(clean_profile):
+    r = run_ffpass_cmd('export', clean_profile('firefox-84'))
     r.check_returncode()
-    assert r.stdout == EXPECTED_EXPORT_OUTPUT
+    assert stdout_splitter(r.stdout) == EXPECTED_EXPORT_OUTPUT
 
 
-def test_legacy_firefox():
-    r = run_ffpass('import', 'tests/firefox-70')
+def test_firefox_aes_export(clean_profile):
+    # This uses your new AES-encrypted profile
+    profile_path = clean_profile('firefox-146-aes')
+    r = run_ffpass_cmd('export', profile_path)
     r.check_returncode()
+    assert stdout_splitter(r.stdout) == EXPECTED_EXPORT_OUTPUT
+
 
-    r = run_ffpass('export', 'tests/firefox-70')
+def test_legacy_firefox(clean_profile):
+    profile_path = clean_profile('firefox-70')
+
+    # modifies the temp file, not the original
+    r = run_ffpass_cmd('import', profile_path)
```

Comment:
This is an important fix imo. It's best not to modify files in the source tree that are used for tests!

Someone could accidentally commit changes and break it.

Better to use a temp copy.